decision-records/006-javascript-compatibility.md

```diff
@@ -0,0 +1,161 @@
+# JavaScript browser compatibility
+
+Date: 27/06/23
+
+Status: Accepted
+
+## Decision
+
+We will ship JavaScript that can parse without errors in browsers supporting `<script type="module">`.
+
+We will update our guidance to tell service teams to include the JavaScript from GOV.UK Frontend using `<script type="module">`.
+
+We will set up:
+
+- transpiling to transform modern code down to that baseline
+- linting to prevent us from using newer features and syntaxes
+
+Going forward, when we want to use a newer feature we will disable the linting rule only after checking the performance impact of the transformation.
```

Comment:
Can potentially scrub this line if we clarify it above?

```suggestion
```

Reply:
I kept it as I think it's important to highlight we decided we won't opt in to things "just because", but have first a check on their impact.